### PR TITLE
feat(search): add algolia search

### PR DESCRIPTION
**Pull request by lahabana**
Get everything setup for algolia search but doesn't show the search box yet
We will display the search box as soon as the index is up to date

We add docsearch meta for the section and the version of the docs
this way all versions are indexed but we can set the search to only search
the blog and the latest version

We also remove a lot of old stuff that wasn't used anymore

Part of #600

Signed-off-by: Charly Molter <charly.molter@konghq.com>

### CODE REVIEW

File: docs/.vuepress/config.js [hunk 1]
```diff
@@ -55,11 +55,14 @@ module.exports = {
...
     editLinks: false,
     sidebarDepth: 0,
-    search: true,
+    // TODO: Reenable as soon as the doc search index is good with meta
+    search: false,
     searchMaxSuggestions: 10,
     algolia: {
-      apiKey: "",
-      indexName: ""
+      apiKey: "4224b11bee5bf294f73032a4988a00ea",
```
Comment (michaelbeaumont) [inline docs/.vuepress/config.js:62]: I suppose this is a properly restricted key that can be made public

**Review by michaelbeaumont — APPROVED** [review]
Will we wait to merge until the index is ready? It'd be nice to see it in netlify preview before merging

**Comment by lahabana** [comment]
We need to merge this as we need to rebuild an index with the meta tags that are added here.
I've hidden the search box ftm and will follow up with a pr to display it once the index is good

**Comment by michaelbeaumont** [comment]
Ah ok got it, it needs to be merged for the indexing to occur :+1:

File: docs/.vuepress/config.js (hunk 1 above)
Comment (lahabana) [reply docs/.vuepress/config.js:62]: Yes that's the instructions they gave me in their intro email:

JavaScript
Copy this snippet at the end of the HTML <body> tag
<script src="https://cdn.jsdelivr.net/npm/@docsearch/js@3"></script>
 <script type="text/javascript">
 docsearch({
 appId: RSEEOBCB49,
 apiKey: 4224b11bee5bf294f73032a4988a00ea,
 indexName: kuma,
 container: '### REPLACE ME WITH A CONTAINER (e.g. div) ###'
 debug: false // Set debug to true if you want to inspect the modal
 });
 </script>

**Comment by jakubdyszkiewicz** [comment]
sweet!!!

**Pull request merged**